provisioning_client/deps/CMakeLists.txt

```diff
@@ -14,51 +14,81 @@ else()
         set(msr_riot_c_files
-            ./RIoT/Emulator/DICE/DiceSha256.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotAes128.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotAesTables.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotCrypt.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotDerEnc.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotEcc.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotHmac.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotKdf.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotSha256.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotX509Bldr.c
-            ./RIoT/Emulator/RIoT/RIoTCrypt/RiotBase64.c
+            ./RIoT/Simulation/DICE/DiceSha256.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/RiotCrypt.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/aes.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/base64.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/bignum.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/derenc.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecdh.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecdsa.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecp.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecp_curves.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/hkdf.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/hmac_drbg.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/md.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/platform_util.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/sha1.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/sha256.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/x509bldr.c
         )
 
         set(msr_riot_h_files
-            ./RIoT/Emulator/DICE/DiceSha256.h
-            ./RIoT/Emulator/RIoT/RIoT.h
-            ./RIoT/Emulator/RIoT/RiotStatus.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotAes128.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotCrypt.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotDerEnc.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotEcc.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotHmac.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotKdf.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotSha256.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotTarget.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotX509Bldr.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotBase64.h
+            ./RIoT/Simulation/DICE/DiceSha256.h
+            ./RIoT/Simulation/RIoT/Core/RIoT.h
+            ./RIoT/Simulation/RIoT/Core/RiotStatus.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/RiotCrypt.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/base64.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/derenc.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/x509bldr.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/aes.h
```

Comment:
How do we specify where to pick up mbedTLS? Does this work only on Linux or can it be used on Windows with a vcpkg installation of mbedTLS?

Reply:
We don't actually pick up mbedtls, RIoT is based on the mbedtls crypto, if you flag use prov_client and hsm_type_x509 windows will pull this in, the e2e test is only run in a linux enviro, however it will run in windows if you adjust the CMakeList